worksheet_D/oxo.py

```diff
@@ -0,0 +1,117 @@
+class OxoBoard:
+    def __init__(self, x, y):
+        self.x = x
+        self.y = y
+        self.board = {(0, 0): 0,(1, 0): 0, (2, 0): 0,
+                      (0, 1): 0, (1, 1): 0, (2, 1): 0,
+                      (0, 2): 0, (1, 2): 0, (2, 2): 0}      # Creates dictionary.
+
+    def get_square(self, x, y):
+        return self.board[x, y]                             # Looks up value with the key (x, y) and returns it
+
+    def set_square(self, x, y, mark):
+        if self.board[x,y] == 0:
+            self.board[x, y] = mark                         # Changes the value assigned to the key to 'mark'
+            return True
+        else:
+            return False
+
+    def is_board_full(self):
```

Comment:
This function always needs to examine every square on the board. Is there a quicker way?